addons/stock_account/wizard/stock_valuation_layer_revaluation.py

```diff
@@ -0,0 +1,139 @@
+            added_value_remaining -= remaining_value_to_add
+        if not float_is_zero(added_value_remaining, precision_rounding=self.currency_id.rounding):
+            remaining_svls[-1].remaining_value += added_value_remaining
+
+        stock_valuation_layer = self.env['stock.valuation.layer'].sudo().create(svl_vals)
+
+        # If the Inventory Valuation of the product category is automated, create related account move.
+        if self.property_valuation == 'real_time':
+            if not self.env['stock.valuation.layer'].check_access_rights('read', raise_exception=False):
+                raise UserError(_("You cannot revalue the stock for product with automated valuation as it leads to the creation of a journal entry, for which you don't have the access rights."))
+
+            accounts = self.product_id.product_tmpl_id.get_product_accounts()
+
+            if self.added_value < 0:
+                debit_account_id = self.account_id.id
+                credit_account_id = accounts.get('stock_valuation') and accounts['stock_valuation'].id
+            else:
+                debit_account_id = accounts.get('stock_valuation') and accounts['stock_valuation'].id
+                credit_account_id = self.account_id.id
+
+            move_vals = {
+                'journal_id': self.account_journal_id.id or accounts['stock_journal'].id,
+                'company_id': self.company_id.id,
+                'ref': _("Revaluation of %s") % self.product_id.display_name,
+                'stock_valuation_layer_ids': [(6, None, [stock_valuation_layer.id])],
+                'date': self.date or fields.Date.today(),
+                'move_type': 'entry',
+                'line_ids': [(0, 0, {
+                    'name': _('%s changed stock valuation from %s to %s of %s') % (
+                        self.env.user.name, self.current_value_svl, self.current_value_svl + self.added_value, self.product_id.display_name),
+                    'account_id': debit_account_id,
+                    'debit': abs(self.added_value),
+                    'credit': 0,
+                    'product_id': self.product_id.id,
+                }), (0, 0, {
+                    'name': _('%s changed stock valuation from %s to %s of %s') % (
+                        self.env.user.name, self.current_value_svl, self.current_value_svl + self.added_value, self.product_id.display_name),
+                    'account_id': credit_account_id,
+                    'debit': 0,
+                    'credit': abs(self.added_value),
+                    'product_id': self.product_id.id,
+                })],
+            }
+            account_move = self.env['account.move'].sudo().create(move_vals)
+            account_move.post()
+
+        # Update the stardard price in case of AVCO
+        if self.product_id.categ_id.property_cost_method == 'average':
+            self.product_id.with_context(disable_auto_svl=True).write({'standard_price': self.new_value_by_qty})
```

Comment:
when applying a landed cost, we don't use this formula, we use
``` python
product.with_context(force_company=self.company_id.id).sudo().standard_price += cost_to_add / product.quantity_svl
```

but i agree they should be equivalent, but in practice they're not

i also thought about just doing product.value_svl  / product.quantity_svl, that should also be equivalent, but in practice it's not the case -_-